core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -478,20 +344,7 @@ protected final void copyBitcoinHeaderTo(final Block block) {
      */
     @Override
     public String toString() {
-        StringBuilder s = new StringBuilder();
-        s.append(" block: \n");
-        s.append("   hash: ").append(getHashAsString()).append('\n');
-        s.append("   version: ").append(version);
-        String bips = Joiner.on(", ").skipNulls().join(isBIP34() ? "BIP34" : null, isBIP66() ? "BIP66" : null,
-                isBIP65() ? "BIP65" : null, isBIP101() ? "BIP101" : null);
-        if (!bips.isEmpty())
-            s.append(" (").append(bips).append(')');
-        s.append('\n');
-        s.append("   previous block: ").append(getPrevBlockHash()).append("\n");
-        s.append("   merkle root: ").append(getMerkleRoot()).append("\n");
-        s.append("   time: [").append(time).append("] ").append(Utils.dateTimeFormat(time * 1000)).append("\n");
-        s.append("   difficulty target (nBits): ").append(difficultyTarget).append("\n");
-        s.append("   nonce: ").append(nonce).append("\n");
+        StringBuilder s = super.toStringBuilder();
```

Comment:
I think a better pattern would be a `void appendToString(StringBuilder b)` in the super class that appends to the given builder.